### ISSUE TITLE
Migrate assignment submission logic into experiment.py: 3pts

### ISSUE THREAD

**Issue by ccs-ucb** [Dallinger/Dallinger#5331]
Currently any participant who fails is immediately re-recruited. There are situations where this is not desirable. It would be great if we could have a method of the experiment class that controls this behavior.

**Comment by pmcharrison** [comment]
Yes, strangely there are a couple of places in the Dallinger code where, when a participant is failed, `experiment.recruiter.recruit` is called instead of `experiment.recruit`, which bypasses any custom logic in the Experiment class. Would you be happy if we fixed those cases to use `experiment.recruit`, so you can put custom logic there to skip recruiting if you want?
